contracts/Migration/MantleTokenMigratorV2.sol

```diff
@@ -0,0 +1,409 @@
+    ///     - The caller must have the Blacklister role
+    /// @param _address The address to blacklist
+    function blacklistAddress(address _address) public onlyBlacklister {
+        blacklistedAddresses[_address] = true;
+
+        emit AddressBlacklisted(msg.sender, _address);
+    }
+
+    /// @notice Removes an address from the blacklist
+    /// @dev emits an {AddressRemovedFromBlacklist} event
+    /// @dev Requirements:
+    ///     - The caller must have the Blacklister role
+    /// @param _address The address to remove from the blacklist
+    function removeAddressFromBlacklist(address _address) public onlyBlacklister {
+        blacklistedAddresses[_address] = false;
+
+        emit AddressRemovedFromBlacklist(msg.sender, _address);
+    }
+
+    /// @notice Halts the contract, preventing token migrations
+    /// @dev emits a {ContractHalted} event
+    /// @dev Requirements:
+    ///     - The caller must be the contract owner
+    function haltContract() public onlyOwner {
+        halted = true;
+
+        emit ContractHalted(msg.sender);
+    }
+
+    /// @notice Unhalts the contract, allowing token migrations
+    /// @dev emits a {ContractUnhalted} event
+    /// @dev Requirements:
+    ///     - The caller must be the contract owner
+    function unhaltContract() public onlyOwner {
+        halted = false;
+
+        emit ContractUnhalted(msg.sender);
+    }
+
+    /// @notice Sets the treasury address
+    /// @dev emits a {TreasuryChanged} event
+    /// @dev Requirements:
+    ///     - The caller must be the contract owner
+    function setTreasury(address _treasury) public onlyOwner {
+        if (_treasury == address(0)) {
+            revert MantleTokenMigrator_InvalidTreasury(_treasury);
+        }
+
+        emit TreasuryChanged(treasury, _treasury);
+
+        treasury = _treasury;
+    }
+
+    // Token Management Functions
+
+    /// @notice Defunds the contract by transferring a specified amount of BIT or MNT tokens to the treasury address
+    /// @dev emits a {ContractDefunded} event
+    /// @dev Requirements:
+    ///     - The caller must be the contract owner
+    ///     - {_tokenAddress} must be either the BIT or the MNT token address
+    ///     - The contract must have a balance of at least {_amount} of {_tokenAddress} tokens
+    /// @param _tokenAddress The address of the token to defund
+    /// @param _amount The amount of tokens to defund
+    function defundContract(address _tokenAddress, uint256 _amount) public onlyOwner {
+        if (_tokenAddress != BIT_TOKEN_ADDRESS && _tokenAddress != MNT_TOKEN_ADDRESS) {
+            revert MantleTokenMigrator_InvalidFundingToken(_tokenAddress);
+        }
+
+        // we can only defund BIT or MNT into the predefined treasury address
+        ERC20(_tokenAddress).safeTransfer(treasury, _amount);
+
+        emit ContractDefunded(treasury, _tokenAddress, _amount);
+    }
+
+    /// @notice Sweeps a specified amount of tokens to an arbitrary address
```

Comment:
suggestion: maybe add a note here to say this is for accidental transfers

Reply:
done